### PR TITLE
Upgrade Zuul to latest

### PR DESCRIPTION
**Pull request by YustinaKvr**
1) Zuul version 9.5.0
2) python-builder:3.11-bookworm
3) node:16-bookworm
4) OpenShift client 4.11.20

### CODE REVIEW

File: docker/zuul/Dockerfile [hunk 1]
```diff
@@ -23,24 +23,24 @@ RUN  git config --global user.email "zuul@otc-service.com" && \
 
 RUN git clone https://opendev.org/zuul/zuul zuul_src && \
     cd /zuul_src && \
-    git pull https://review.opendev.org/zuul/zuul refs/changes/40/859940/10
+    git pull https://review.opendev.org/zuul/zuul refs/changes/22/908722/1
```
Comment (vladimirhasko) [inline docker/zuul/Dockerfile:26]: what is this PR and why not referring to custom zuul change provided by @gtema ?

https://review.opendev.org/c/zuul/zuul/+/859940

**Review by vladimirhasko — CHANGES_REQUESTED** [review]

**Comment by gtema** [comment]
such change will immediately drop support for gitea with no rollback

File: docker/zuul/Dockerfile (hunk 1 above)
Comment (YustinaKvr) [reply docker/zuul/Dockerfile:26]: This pr from https://opendev.org/zuul/zuul/src/commit/81d3c861153ee1fc9709d0fd8e9c2f8bce96ee7a it linked with 9.5.0 version, next major version from 8.2.1

Is this ref should be static?

**Review by gtema — CHANGES_REQUESTED** [review]
1. You can't simply upgrade zuul unless the mentioned change (points to the https://review.opendev.org/c/zuul/zuul/+/859939/15) is merged. You should overtake it and maintain until it gets merged (including promoting in the zuul community necessity to get it landed). This change implements gitea support
2. all other things in the dockerfile are copied from https://opendev.org/zuul/zuul/src/branch/master/Dockerfile and you should keep them in sync with the version of zuul you want to have running to prevent failures

**Review by vladimirvshivkov — APPROVED** [review]

File: docker/zuul/Dockerfile (hunk 1 above)
Comment (vladimirvshivkov) [reply docker/zuul/Dockerfile]: resolved

**Comment by vladimirvshivkov** [comment]
@gtema

> 1. You can't simply upgrade zuul unless the mentioned change (points to the https://review.opendev.org/c/zuul/zuul/+/859939/15) is merged. You should overtake it and maintain until it gets merged (including promoting in the zuul community necessity to get it landed). This change implements gitea support
> 2. all other things in the dockerfile are copied from https://opendev.org/zuul/zuul/src/branch/master/Dockerfile and you should keep them in sync with the version of zuul you want to have running to prevent failures

we use your latest pr for now.

**Comment by vladimirvshivkov** [comment]
d

**Review by gtema — APPROVED** [review]
Please note also that you will need to use corresponding nodepool version and please do some testing beforehand - unittests are now passing (failing ones are not related to the change) but sich a major upgrade is not without risks.

**Review by vladimirhasko — APPROVED** [review]

**Pull request merged**